server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -275,26 +271,15 @@ private void updateMailbox(Mailbox mailbox, MailboxUpdateRequest updateRequest,
         }
 
         if (!originMailboxPath.equals(destinationMailboxPath)) {
-            for (MailboxMetaData mailboxMetaData: originalMailboxPathWithChild) {
-                subscriptionManager.unsubscribe(mailboxSession, mailboxMetaData.getPath().getName());
-            }
+            List<MailboxRenamedResult> mailboxRenamedResults = mailboxManager.renameMailbox(mailbox.getId(), destinationMailboxPath, mailboxSession);
 
-            mailboxManager.renameMailbox(mailbox.getId(), destinationMailboxPath, mailboxSession);
-            List<MailboxMetaData> destinationMailboxPathWithChild = findMailboxMetadataListIncludeChild(destinationMailboxPath, mailboxSession);
-            for (MailboxMetaData mailboxMetaData: destinationMailboxPathWithChild) {
-                subscriptionManager.subscribe(mailboxSession, mailboxMetaData.getPath().getName());
+            for (MailboxRenamedResult mailboxRenamedResult: mailboxRenamedResults) {
+                subscriptionManager.unsubscribe(mailboxSession, mailboxRenamedResult.getOriginPath().getName());
+                subscriptionManager.subscribe(mailboxSession, mailboxRenamedResult.getDestinationPath().getName());
```

Comment:
I would extract these two line into an `udateSubscriptions` method and use a foreach.